### DISCUSSION TITLE
Keyboard shortcut to open file manager of currently played media

### DISCUSSION THREAD

**Discussion by lamyergeier** [mpv-player/mpv#14084] [Q&A] [answered]
Is it possible to define the "Keyboard shortcut to open file manager of currently played media". I use Gnome 46, so I would like to open nautilus.

**Comment by ruihe774** [comment]
In `input.conf`:

```
ctrl+o run "/bin/bash" "-c" "gio open `dirname ${path}`"
```

**Reply by lamyergeier** [reply, marked as answer]
@ruihe774 I get the following error in terminal when I press the shortcut

```
/bin/bash: command substitution: line 1: syntax error near unexpected token `('
/bin/bash: command substitution: line 1: `dirname Don Giovanni - Commendatore Scene - EN Sub (Better Quality) [Ioc9shJa_lI].webm'
gio: No locations given

Usage:
  gio open LOCATION…

Open files with the default application that
is registered to handle files of this type.
```

# Solution

I added quotation marks

```
ctrl+o run "/bin/bash" "-c" "gio open \"$(dirname \"${path}\")\""
```

**Reply by lamyergeier** [reply]
@ruihe774 Is it also possible to display a status message (for like 5 seconds) on top left of the mpv window like "Opening in Nautilus" along with executing the action of opening it ?

**Reply by ruihe774** [reply]
https://mpv.io/manual/master/#command-interface-show-text

**Reply by lamyergeier** [reply]
@ruihe774 

This worked successfully.

The following works when I run MPV from terminal, but when I click on video file in Gnome file browser and open in MPV from GUI, it doesn't work.

Goal: Open the youtube url stored in `*.info.json` file in the same directory as the video. I read jq to read the url.

I added the following in input.conf

```
# Open the video in youtube
ctrl+y run "/bin/bash" "-c" "/usr/local/bin/firefox $(/usr/bin/jq -r .webpage_url \"$(dirname \"${path}\")\"/*.info.json)"; show-text "Open in firefox" 2000
```

For example the a.info.json file  is

```json
{
	"webpage_url": "https://www.youtube.com/watch?v=eFoNOFVBCl0"
}
```

### Issue: Why this works from Terminal but not when mpv is started by clicking on the Video file in Gnome Files?

**Reply by ruihe774** [reply]
> but when I click on video file in Gnome file browser and open in MPV from GUI, it doesn't work.

Because Nautilus (GNOME file browser) uses portal or GVFS; you can have a look at the value of `${path}` and it is not the original path of the file you click in Nautilus.

**Reply by lamyergeier** [reply]
> Because Nautilus (GNOME file browser) uses portal or GVFS; you can have a look at the value of `${path}` and it is not the original path of the file you click in Nautilus.

Does it mean that it is not possible to use a keyboard shortcut involving `path` both from Gnome files and from Terminal? But how does the following works from Gnome Files and also from Terminal

```
# Open the video folder in file manager
ctrl+o run "/bin/bash" "-c" "gio open \"$(dirname \"$(dirname \"${path}\")\")\""; show-text "Open in file manager" 2000
```

**Reply by ruihe774** [reply]
Sorry, I have no idea. You can try debugging parts of your command. e.g. checking if `\"$(dirname \"${path}\")\"/*.info.json` and `$(/usr/bin/jq -r .webpage_url \"$(dirname \"${path}\")\"/*.info.json)` have expected values by `echo` them.

**Reply by lamyergeier** [reply]
It works fine when I run mpv from terminal. I don't know how to debug when I am clicking video file on Nautilus as there is no place to see the error.

**Reply by ruihe774** [reply]
If you cannot see the output of `echo` giving there is no terminal, you can try `notify-send`

**Reply by blueray453** [reply]
i use the following lua script:

```
function open_in_file_manager()
    local path = mp.get_property("path")
    if not path then
        mp.osd_message("No file loaded")
        return
    end

    os.execute(string.format('nemo "%s" &', path))
    -- mp.osd_message("Opened in file manager:\n" .. path)
end

mp.add_key_binding("ctrl+d", "open_in_file_manager", open_in_file_manager)
```

I use it for nemo. You have to change it for your file browser.
